wagtail/admin/viewsets/model.py

```diff
@@ -89,35 +132,111 @@ def index_results_view(self):
+        attributes on the class.
+        """
+        if not self.template_prefix:
+            return [fallback]
+        templates = [
+            f"{self.template_prefix}{self.app_label}/{self.model_name}/{action}.html",
+            f"{self.template_prefix}{self.app_label}/{action}.html",
+            f"{self.template_prefix}{action}.html",
+        ]
+        if fallback:
+            templates.append(fallback)
+        return templates
+
+    @cached_property
+    def index_template_name(self):
+        """
+        A template to be used when rendering ``index_view``.
+
+        Default: if :attr:`template_prefix` is specified, an ``index.html``
+        template in the prefix directory and its ``{app_label}/{model_name}/``
+        or ``{app_label}/`` subdirectories will be used. Otherwise, the
+        ``index_view_class.template_name`` will be used.
+        """
+        return self.get_templates(
+            "index",
+            fallback=self.index_view_class.template_name,
+        )
+
+    @cached_property
+    def index_results_template_name(self):
+        """
+        A template to be used when rendering ``index_results_view``.
+
+        Default: if :attr:`template_prefix` is specified, a ``index_results.html``
+        template in the prefix directory and its ``{app_label}/{model_name}/``
+        or ``{app_label}/`` subdirectories will be used. Otherwise, the
+        ``index_view_class.results_template_name`` will be used.
+        """
+        return self.get_templates(
+            "index_results",
+            fallback=self.index_view_class.results_template_name,
+        )
+
+    @cached_property
+    def create_template_name(self):
+        """
+        A template to be used when rendering ``add_view``.
+
+        Default: if :attr:`template_prefix` is specified, a ``create.html``
+        template in the prefix directory and its ``{app_label}/{model_name}/``
+        or ``{app_label}/`` subdirectories will be used. Otherwise, the
+        ``add_view_class.template_name`` will be used.
+        """
+        return self.get_templates(
+            "create",
+            fallback=self.add_view_class.template_name,
+        )
+
+    @cached_property
+    def edit_template_name(self):
+        """
+        A template to be used when rendering ``edit_view``.
+
+        Default: if :attr:`template_prefix` is specified, an ``edit.html``
+        template in the prefix directory and its ``{app_label}/{model_name}/``
+        or ``{app_label}/`` subdirectories will be used. Otherwise, the
+        ``edit_view_class.template_name`` will be used.
+        """
+        return self.get_templates(
+            "edit",
+            fallback=self.edit_view_class.template_name,
+        )
+
+    @cached_property
+    def delete_template_name(self):
+        """
+        A template to be used when rendering ``delete_view``.
+
+        Default: if :attr:`template_prefix` is specified, a ``delete.html``
```

Comment:
Not sure if we should expect a `delete.html` or a `confirm_delete.html` here. The generic views use `confirm_delete.html` but I find it inconsistent with the other views.

Reply:
Yep, that makes sense. I've renamed it to `name` in 3cb897ca49572f0495f2573a510005e8af634209, because it can even technically be a path relative to the `template_prefix`. The `action` was a leftover from ModelAdmin. Not sure if that warrants an upgrade consideration though, as it's simple enough.